include/maliput/geometry_base/road_geometry.h

```diff
@@ -131,7 +132,16 @@ class RoadGeometry : public api::RoadGeometry {
 
   ~RoadGeometry() override = default;
 
+  void SpacialReorganization(const SpacialReorganization::Type& type);
```

Comment:
I think we should review this approach.
I suggest an approach like this: https://coliru.stacked-crooked.com/a/d7b00994ff8dec57

It will require templetizing this class and defining a default implementation for it with the kd tree. We are not likely going to change the approach from using kd-tree to another implementation (ie. brute force).